clickup_integration/clickup_integration/doctype/clickup_settings/clickup_settings.py

```diff
@@ -0,0 +1,264 @@
+# Copyright (c) 2024, Parsimony LLC and contributors
+# For license information, please see license.txt
+
+import frappe
+import urllib
+import requests
+from frappe.model.document import Document
+from frappe.desk.form.assign_to import add as assign_task
+from clickup_integration.constants import (
+	GET_TEAMS,
+	GET_SPACES,
+	GET_FOLDERS,
+	GET_TASKS,
+	GET_TASK,
+	GET_LISTS,
+	GET_COMMENTS
+)
+
+
+class ClickupSettings(Document):
+	def get_teams(self):
+		if not self.get_password("access_token"):
+			frappe.throw("No Access token found please authorize.")
+		response = requests.get(
+			GET_TEAMS,
+			headers= {
+				"Content-Type": "application/json",
+				"Authorization": self.get_password("access_token")
+			}
+		)
+		if response.status_code == 200:
+			response = frappe.parse_json(response.content.decode())
+			for team in response.get("teams"):
+				self.get_spaces(team_id=team.get("id"))
+
+	def get_spaces(self, team_id):
+		response = requests.get(
+			GET_SPACES.format(team_id=team_id),
+			headers= {
+				"Content-Type": "application/json",
+				"Authorization": self.get_password("access_token")
+			}
+		)
+		if response.status_code == 200:
+			response = frappe.parse_json(response.content.decode())
+			for space in response.get("spaces"):
+				self.get_folders(space_id=space.get("id"))
+
+	def get_folders(self, space_id):
+		response = requests.get(
+			GET_FOLDERS.format(space_id=space_id),
+			headers= {
+				"Content-Type": "application/json",
+				"Authorization": self.get_password("access_token")
+			}
+		)
+		if response.status_code == 200:
+			response = frappe.parse_json(response.content.decode())
+			for folder in response.get("folders"):
+				project = frappe.db.exists("Project", {"project_name": folder.get("name")})
+				if not project:
+					project_doc = frappe.new_doc("Project")
+					project_doc.update({
+						"project_name": folder.get("name")
+					})
+					project_doc.save()
+					project = project_doc.name
+					frappe.db.commit()
+				self.get_lists(folder_id=folder.get("id"), project=project)
+
+	def get_lists(self, folder_id, project=None):
```

Comment:
Project can't be optional here, right? Since `get_tasks` needs a project value

```suggestion
	def get_lists(self, folder_id, project):
```